spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -743,9 +743,7 @@ public void run() {
 			if (this.genericListener instanceof ConsumerSeekAware) {
 				((ConsumerSeekAware) this.genericListener).registerSeekCallback(this);
 			}
-			if (this.transactionManager != null) {
-				ProducerFactoryUtils.setConsumerGroupId(this.consumerGroupId);
-			}
+			KafkaUtils.setConsumerGroupId(this.consumerGroupId);
```

Comment:
Looks like there is no a conditions to not set this value into the `ThreadLocal` any more...